src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,2356 @@
+                    case 6989:
+                        if (!player->HasAura(SPELL_CURSE_OF_THE_SKUNK))
+                            player->CastSpell(player, SPELL_CURSE_OF_THE_SKUNK, true);
+                        break;
+                    case 6990:
+                        if (!player->HasAura(SPELL_CURSE_OF_THE_CROCODILE))
+                            player->CastSpell(player, SPELL_CURSE_OF_THE_CROCODILE, true);
+                        break;
+                    case 6991:
+                    case 6992:
+                        if (!player->HasAura(SPELL_CURSE_OF_THE_CRANE))
+                            player->CastSpell(player, SPELL_CURSE_OF_THE_CRANE, true);
+                        break;
+                    case 7011:
+                    case 7012:
+                        if (!player->HasAura(SPELL_CURSE_OF_THE_TURTLE))
+                            player->CastSpell(player, SPELL_CURSE_OF_THE_TURTLE, true);
+                        break;
+                }
+            }
+            else
+            {
+                switch (areaTrigger->ID)
+                {
+                    case 6986:
+                    case 6987:
+                        if (!player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6986)) && !player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6987)))
+                            if (player->HasAura(SPELL_CURSE_OF_THE_FROG))
+                                player->RemoveAura(SPELL_CURSE_OF_THE_FROG);
+                        break;
+                    case 6988:
+                    case 6989:
+                        if (!player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6988)) && !player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6989)))
+                            if (player->HasAura(SPELL_CURSE_OF_THE_SKUNK))
+                                player->RemoveAura(SPELL_CURSE_OF_THE_SKUNK);
+                        break;
+                    case 6990:
+                        if (player->HasAura(SPELL_CURSE_OF_THE_CROCODILE))
+                            player->RemoveAura(SPELL_CURSE_OF_THE_CROCODILE);
+                        break;
+                    case 6991:
+                    case 6992:
+                        if (!player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6991)) && !player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(6992)))
+                            if (player->HasAura(SPELL_CURSE_OF_THE_CRANE))
+                                player->RemoveAura(SPELL_CURSE_OF_THE_CRANE);
+                        break;
+                    case 7011:
+                    case 7012:
+                        if (!player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(7011)) && !player->IsInAreaTriggerRadius(sAreaTriggerStore.LookupEntry(7012)))
+                            if (player->HasAura(SPELL_CURSE_OF_THE_TURTLE))
+                                player->RemoveAura(SPELL_CURSE_OF_THE_TURTLE);
+                        break;
```

Comment:
Cleanup those `LookupEntry` calls, you already have the correct AT object.
`new at_singing_pools_transform(<area-trigger-id>, <spell-to-apply))`

Alternatively, this could be done using a combination of `spell_area_aura` and `conditions` (pretty sure we have areatrigger conditions)

Reply:
First of all, thanks for review.
I have only one AT object at the time it was triggered.
But there are two overlapping areatriggers in each pool, and I need to determine whether the player left both of them or not, meaning, if I trigger (leave) one areatrigger, I need to know if player is out of the second areatrigger too.

![image](https://user-images.githubusercontent.com/32191/62491234-8ccd0000-b7cb-11e9-8f27-463071e61c17.png)

Reply:
Well at least one of those lookups is redundant in each case then. Also change them to `AssertLookup`, we have that now

Reply:
done